Fix for #193: Content Explorer does not show Uploader name

sghosh326 opened the pull request:
- moved the FIELD_MODIFIED_BY from SIDEBAR_FIELDS_TO_FETCH to BASE_FIELDS_TO_FETCH and adjusted the fields-test.js so tests pass.
- Added ItemModified component to output the Modified by name in the modified column
- Added modifiedCellRenderer to render the Modified column cells
- Updated ItemList to use modifiedCellRenderer instead of default Date cell renderer. Increased width of Modified column to 300px.

boxcla commented:
Verified that @sghosh326 has signed the CLA. Thanks for the pull request!

priyajeet commented on src/components/ContentExplorer/ItemList.js:
```diff
@@ -13,14 +13,14 @@ import 'react-virtualized/styles.css';
 import messages from '../messages';
 import {
-    FIELD_NAME,
+    FIELD_NAME,   
```
Extra whitespace after ,

priyajeet commented on src/components/ContentExplorer/ItemList.js:
```diff
@@ -89,8 +89,11 @@ const ItemList = ({
         isTouch
     );
+    const modifiedCell = modifiedCellRenderer(
+    	rootId,
```
I think having them in 1 line should not give eslint errors.
If you have eslint and prettier configured, that will re-arrange the file.

sghosh326 replied:
I have used the same formatting as other cell renderers. As for eslint, I installed it and configured it to use the Air BnB style as in package.json. The .eslintrc.json has this:
{
    "extends": "airbnb"
}
I get a dependency resolutions error when I attempt to run eslint for this file.
The Travis CI build error were with the ItemModified.js and modifiedCellRenderer.js files. I am fixing the errors on those files. 
Is there a build tool that I can run that will report these errors?

priyajeet commented on src/components/Item/ItemModified.js:
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
```
Can you add flow/doc on the top of this file, like its there for other js files

sghosh326 replied:
I have added flow/doc comment

priyajeet commented on src/components/Item/ItemModified.js:
```diff
@@ -0,0 +1,28 @@
+    const { modified_at = '', interacted_at = '', modified_by }: BoxItem = item;
+    const isRecents: boolean = view === VIEW_RECENTS;
+    const date: string = isRecents ? interacted_at || modified_at : modified_at;
+    const modified_by_name: string = modified_by.name;
```
camel casing for declared variable/const names, no _s.

priyajeet commented on src/components/Item/modifiedCellRenderer.js:
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
```
Same flow doc on top

sghosh326 replied:
added

priyajeet commented on src/components/Item/modifiedCellRenderer.js:
```diff
@@ -0,0 +1,17 @@
+    rootId: string,
+    view: View,
+) => ({ rowData }: { rowData: BoxItem }) => (
+    <div className='be-item-modified'>
```
Is this wrapping div needed? Can this div be just moved inside ItemModified and used in-place of the wrapping span that has? Or just the span used without div?

priyajeet commented on src/components/Item/ItemModified.js:
```diff
@@ -0,0 +1,28 @@
+
+    return (
+   		<span className='be-item-label'>
+            <Datefield date={date} />&nbsp;{byLabel}&nbsp;{modified_by_name}
```
Probably should not try to concatenate a line based on localization, because what maybe fine in one language, might be different in another. So doing `date by name` may not be written/spoken in that order in another locale.

Also, since its in-dom style, should use <FormattedMessage> in-line as opposed to intl, whenever possible.

So instead do it something like the rich text formatting example as shown here https://github.com/yahoo/react-intl/wiki/Components#formattedmessage

```
<FormattedMessage
    {...messages.modifiedColumnValue}
    values={{
        date: <Datefield date={date} />
        name: modifiedByName
    }}
/>
```

```
modifiedColumnValue: {
        id: 'be.modifiedColumnValue',
        description: 'Text for modified column',
        defaultMessage: '{date} by {name}:'
    },
```

priyajeet replied:
The other thing I will check, is that can the name be empty at times. Since UI is rendered based on auth token scope and does that get a name in all cases. If it doesn't we may have to drop the "by name" part for those cases or make it like `by Unknown` or something.

priyajeet commented on src/components/ContentExplorer/ItemList.js:
```diff
@@ -187,9 +190,9 @@ const ItemList = ({
+                                    cellRenderer={modifiedCell}
                                     headerRenderer={headerCellRenderer}
-                                    width={135}
+                                    width={300}
```
May have to see (test around a bit) if the flexing option below needs to change, since its hard to box names in a fixed width. Was easier with just date. So if the explorer size is wide enough, maybe this column should be allowed to expand.

sghosh326 replied:
Should I add flexGrow={1} attribute to the Modified column? It makes the Filename and Modified columns equal width and the flexShrink value will remove the column if the width is too small. It looked good to me while resizing the Content Explorer container.

sghosh326 replied:
added flexGrow={1}

sghosh326 commented:
I have addressed most of your comments from the review in this checkin:
https://github.com/box/box-ui-elements/pull/203/commits/f5956b99e0c6204614f319fdf8ccee6e36bd495f
However after I installed eslint it has messed up the formatting on all the files. I tried to follow the instructions in the eslint docs to set it up. Apparently the eslint styles that were configured are not compatible with the project. You can reject this pull request and take whatever code you think will be useful to add this feature.
Thanks,
-Sandip
